src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -807,7 +807,7 @@ and CheckValUse (cenv: cenv) (env: env) (vref: ValRef, vFlags, m) (ctxt: PermitB
         let isCallOfConstructorOfAbstractType = 
             (match vFlags with NormalValUse -> true | _ -> false) && 
             vref.IsConstructor && 
-            (match vref.DeclaringEntity with Parent tcref -> isAbstractTycon tcref.Deref | _ -> false)
+            (match vref.TryDeclaringEntity with Parent tcref -> isAbstractTycon tcref.Deref | _ -> false)
```

Comment:
We use this "Parent" terminology to mean "DeclaringEntity option".  We should really remove it in favour of that I guess, but I'll leave it in for now as it's quite pervasive in identifiers too